crates/polars-plan/src/dsl/expr.rs

```diff
@@ -67,9 +67,9 @@ impl AsRef<Expr> for AggExpr {
 #[must_use]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum Expr {
-    Alias(Arc<Expr>, Arc<str>),
-    Column(Arc<str>),
-    Columns(Vec<String>),
+    Alias(Arc<Expr>, ColumnName),
+    Column(ColumnName),
+    Columns(Arc<[ColumnName]>),
```

Comment:
Driveby.  Use `ColumnName: Arc<str>` here. That ensures we don't allocate strings down the road and the datatypes between different ways of referring to columns align.